src/iptc.cpp

```diff
@@ -345,22 +346,24 @@ namespace Exiv2 {
 
     void IptcData::printStructure(std::ostream& out, const Slice<byte*>& bytes, uint32_t depth)
     {
-        uint32_t i = 0;
-        while (i < bytes.size() - 3 && bytes.at(i) != 0x1c)
+        size_t i = 0;
+        while (i + 3 < bytes.size() && bytes.at(i) != 0x1c)
             i++;
```

Comment:
The subtraction `bytes.size() - 3` could overflow. `i + 3` cannot, because `i` is initially 0 and only incremented by 1 on each loop iteration.